lib/cc/engine/gem_detector.rb

```diff
@@ -0,0 +1,17 @@
+require "bundler/lockfile_parser"
+
+class GemDetector
+  def initialize(lockfile)
+    @contents = File.read(lockfile)
+  end
+
+  def present?(name)
+    begin
+      @lockfile ||= Bundler::LockfileParser.new(@contents)
+    rescue Bundler::LockfileError
```

Comment:
I think it's safe to move this `rescue` to the end of the method so you can ditch the explicit `begin` and `return`

```rb
def present?(name)
  @lockfile ...
  @lockfile.any? ...
rescue ...
  false
end
```